Lib/selectors.py

```diff
@@ -66,12 +66,19 @@ def __init__(self, selector):
     def __len__(self):
         return len(self._selector._fd_to_key)
 
+    def get(self, fileobj, default=None):
+        fd = self._selector._fileobj_lookup(fileobj)
+        key = self._selector._fd_to_key.get(fd)
+        if key is None:
+            return default
+        return key
+
     def __getitem__(self, fileobj):
-        try:
-            fd = self._selector._fileobj_lookup(fileobj)
-            return self._selector._fd_to_key[fd]
-        except KeyError:
-            raise KeyError("{!r} is not registered".format(fileobj)) from None
+        fd = self._selector._fileobj_lookup(fileobj)
+        key = self._selector._fd_to_key.get(fd)
+        if key is None:
```

Comment:
How about using `self.get(fileobj)` here? Is it too expensive?

Reply:
Assuming https://github.com/python/cpython/issues/106554 merges as-is that would make sense to make it more DRY.

Reply:
Will make that change if #106554 still uses the `.get` in the final outcome.

Reply:
https://github.com/python/cpython/pull/106555/files#diff-671465c62a2c7c951b397550dc8f2f1c03860af4c089299e6edea29347cf6018R465

Reply:
Also with asyncio the `__getitem__` path should almost never be called after https://github.com/python/cpython/pull/106528

Reply:
![_add_writer](https://github.com/python/cpython/assets/663432/db2280f5-d658-437c-980f-e2db70df3876)